api/filter_dimensions.go

```diff
@@ -87,12 +83,12 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 		)
 		return
 	}
-
-	if err := api.isValidDatasetDimensions(ctx, v, []model.Dimension{req.Dimension}, filter.PopulationType); err != nil {
+	dimensions, _, err := api.ValidateCantabularFilter(v, []model.Dimension{req.Dimension}, filter.PopulationType)
```

Comment:
To me the name of function is a bit misleading as it is not validating the filter, rather is running validations and actually populating/returning dimensions.

Reply:
true, will change